api.md

```diff
@@ -0,0 +1,230 @@
+# Table of contents
+- [API](#api)
+    - [Models](#models)
+        - [Project](#project)
+        - [Image](#image)
+        - [Developer](#developer)
+        - [Language](#language)
+    - [Endpoints](#endpoints)
+        - [/projects](#get-projects)
+        - [/project/id](#get-projectintid)
+        - [/developers](#get-developers)
+        - [/developer/id](#get-developerintid)
+
+# API
+[Back to Top](#table-of-contents)
+
+All endpoints will return a json response either acknowledging a POST request or with data relevant to a GET request. Errors should have a status in the 4XX or 5XX depending on the error and should include a 'message' json key with an adequate error message.
+
+## Models
+[Back to Top](#table-of-contents)
+
+All models should be described in the following sections. If implementation requires changes to these models please edit the corresponding information.
+
+### Project
+[Back to Top](#table-of-contents)
+
+Project is a model storing information about a project. It has a unique name, descriptions, icon, many languages, many contributors, a project status, a start year and an end year. This is the suggested django code which might need to be edited. (ID should not be included, it is auto generated)
+
+```py
+id: Implicitely created by django PK
+name = models.CharField(max_length=31, unique=True)
+short_description = models.TextField(max_length=255)
+description_intro = models.TextField(max_length=512)
+description_body = models.TextField(max_length=1024)
+icon = models.ForeignKey(Image, on_delete=models.CASCADE)
+github_link = models.ForeignKey(URL, on_delete=models.CASCADE)
+languages = models.ManyToManyField(Language)
+contributors = models.ManyToManyField(Developer)
+project_status = models.CharField(max_length=1, choices=PROJECT_STATUSES)
+start_year = models.DateField()
+end_year = models.DateField(null=True)
+
+PROJECT STATUSES = {
+	"O": "Ongoing",
+	"U": "Upcoming",
+	"C": "Completed",
+    "H": "On Hold",
+}
+```
+
+### URL
+
+[Back to Top](#table-of-contents)
+
+URL is a model that simply holds a url to enable one to many relationships.
+
+```py
+id: Implicitely generated by django PK
+url = models.URLField(max_length=200)
+```
+
+### Image
+[Back to Top](#table-of-contents)
+
+Stores images.
+
+```
+Implementation pending
+```
+
+### Developer
+[Back to Top](#table-of-contents)
+
+Developer is a model storing information about developers on the website. It has first_name, last_name and github_name. Full names must be unique and github_name must be unique. Devs can have a linkedin url. Devs can have another url of a certain type. The type is used by the frontend to show an icon.
+
+```py
```

Comment:
I say add description fields like project. Just to introduce ourselves and talk about our on-going projects or talk about stuff we like. It would add to the profile instead of just having a picture with a name